src/core/search/search_test.cc

```diff
@@ -907,6 +907,189 @@ TEST_F(SearchTest, NotImplementedSearchTypes) {
       << "Infix search should return a not implemented error";
 }
 
+// Enumeration for different search types
+enum class SearchType { PREFIX = 0, SUFFIX = 1, INFIX = 2 };
+
+// Helper function to generate content with ASCII characters
+static std::string GenerateWordSequence(size_t word_count, size_t doc_offset = 0) {
+  std::string content;
+  for (size_t i = 0; i < word_count; ++i) {
+    std::string word;
+    char start_char = 'a' + ((doc_offset + i) % 26);
+    size_t word_len = 3 + (i % 5);  // Word length 3-7 chars
+
+    for (size_t j = 0; j < word_len; ++j) {
+      char c = start_char + (j % 26);
+      if (c > 'z')
+        c = 'a' + (c - 'z' - 1);
+      word += c;
+    }
+
+    if (i > 0)
+      content += " ";
+    content += word;
+  }
+  return content;
+}
+
+// Helper function to generate pattern with variety
+static std::string GeneratePattern(SearchType search_type, size_t pattern_len,
+                                   bool use_uniform = true) {
+  if (use_uniform) {
+    // Original uniform pattern for comparison
+    switch (search_type) {
+      case SearchType::PREFIX:
+        return std::string(pattern_len, 'p');
+      case SearchType::SUFFIX:
+        return std::string(pattern_len, 's');
+      case SearchType::INFIX:
+        return std::string(pattern_len, 'i');
+    }
+  } else {
+    // Diverse ASCII pattern
+    std::string pattern;
+    char base_char = (search_type == SearchType::PREFIX)   ? 'p'
+                     : (search_type == SearchType::SUFFIX) ? 's'
+                                                           : 'i';
+
+    for (size_t i = 0; i < pattern_len; ++i) {
+      char c = base_char + (i % 10);  // Use variety of chars
+      if (c > 'z')
+        c = 'a' + (c - 'z' - 1);
+      pattern += c;
+    }
+    return pattern;
+  }
+  return "";
+}
+
+// Template function for benchmark implementations
+template <bool UseDiversePattern> static void BM_SearchByTypeImpl(benchmark::State& state) {
```

Comment:
not a good use for templates. Please move it to be a runtime argument.
In general, we do not write templatized code unless there is a good reason to use them.